spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -0,0 +1,199 @@
+      }
+    }
+    getCategoricalFeatureInfo(inputSchema, featuresToInclude, featuresToExclude, predictionField, cardinalityMapping);
+  }
+
+  /**
+   * Get the feature list of the features that have to be used for training/prediction depending on the
+   * featuresToInclude or featuresToInclude list.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training/prediction.
+   * @param featuresToExclude features to be excluded when training/predicting.
+   * @param predictionField   field containing the prediction values.
+   * @return feature list to be used for training/prediction.
+   */
+  protected static Map<String, Integer> getFeatureList(Schema inputSchema, @Nullable String featuresToInclude,
+                                                       @Nullable String featuresToExclude, String predictionField) {
+    if (!Strings.isNullOrEmpty(featuresToExclude) && !Strings.isNullOrEmpty(featuresToInclude)) {
+      throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
+                                           "Please specify fields for one.");
+    }
+    Map<String, Integer> fields = new HashMap<>();
+
+    if (!Strings.isNullOrEmpty(featuresToInclude)) {
+      Iterable<String> tokens = Splitter.on(',').trimResults().split(featuresToInclude);
+      String[] features = Iterables.toArray(tokens, String.class);
+      for (int i = 0; i < features.length; i++) {
+        String field = features[i];
+        Schema.Field inputField = inputSchema.getField(field);
+        if (!field.equals(predictionField) && inputField != null) {
+          fields.put(field, i);
+        }
+      }
+      return fields;
+    }
+
+    Set<String> excludeFeatures = new HashSet<>();
+    if (!Strings.isNullOrEmpty(featuresToExclude)) {
+      excludeFeatures.addAll(Lists.newArrayList(Splitter.on(',').trimResults().split(featuresToExclude)));
+    }
+    Schema.Field[] inputSchemaFields = (Schema.Field[]) inputSchema.getFields().toArray();
+    for (int i = 0; i < inputSchemaFields.length; i++) {
+      String field = inputSchemaFields[i].getName();
+      if (!field.equals(predictionField) && !excludeFeatures.contains(field)) {
+        fields.put(field, i);
+      }
+    }
+    return fields;
+  }
+
+  /**
+   * Get the feature to cardinality mapping provided by the user.
+   *
+   * @param inputSchema        schema of the received record.
+   * @param featuresToInclude  features to be used for training/prediction.
+   * @param featuresToExclude  features to be excluded when training/predicting.
+   * @param labelField         field containing the prediction values.
+   * @param cardinalityMapping feature to cardinality mapping specified for categorical features.
+   * @return categoricalFeatureInfo for categorical features.
+   */
+  protected static Map<Integer, Integer> getCategoricalFeatureInfo(Schema inputSchema,
```

Comment:
This is intended to be used in the plugin transform and run methods.
Hence, keeping this protected